packages/components/src/custom-select-control-v2/legacy-component/index.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+/**
+ * Internal dependencies
+ */
+import _CustomSelect from '../custom-select';
+import type { LegacyCustomSelectProps } from '../types';
+import { CustomSelectItem } from '..';
+
+function _LegacyCustomSelect( props: LegacyCustomSelectProps ) {
+	// Forward props + store from v2 implementation
+	const store = Ariakit.useSelectStore( {
+		async setValue( value ) {
+			if ( ! props.onChange ) return;
+
+			// Executes the logic in a microtask after the popup is closed.
+			// This is simply to ensure the isOpen state matches that in Downshift.
+			await Promise.resolve();
+			const state = store.getState();
+
+			const changeObject = {
+				selectedItem: {
+					// value will always be a string for the legacy component
+					name: value as string,
+					key: state.activeId as string,
+				},
+				highlightedIndex: state.renderedItems.findIndex(
+					( item ) => item.value === value
+				),
+				isOpen: state.open,
+			};
+			props.onChange( changeObject );
+		},
+	} );
+
+	const transformOptionsToChildren = () => {
```

Comment:
Initially, it had some arguments but was modified a bit since, so it makes sense for it to be a constant now. 👍 I think I also gravitate towards functions when returning JSX (not necessarily for any reason other than habit). I am curious if there would be any differences between them, in this case, besides semantics?